pyphi/actual.py

```diff
@@ -413,20 +413,19 @@ def find_causal_link(self, direction, mechanism, purviews=False,
             return CausalLink(max_ria)
         else:
             # Finds rias with maximum alpha
-            all_ria = [self.find_mip(direction, mechanism, purview,allow_neg)
-                          for purview in purviews]
+            all_ria = [self.find_mip(direction, mechanism, purview, allow_neg)
+                       for purview in purviews]
             max_alpha = max(all_ria).alpha
-            max_rias = [ria for ria in all_ria if ria.alpha==max_alpha]
+            max_rias = [ria for ria in all_ria if ria.alpha == max_alpha]
 
             # Selected rias whose purview is not a superset of any other
+            def is_not_superset(purview):
+                return np.all([(not set(purview).issuperset(set(purview2))) or
+                               (set(purview) == set(purview2)) for purview2 in purviews])
+                               
```

Comment:
Will, is this kind of inner function declaration, ok?

Reply:
Fixed, except from the logic part because`is_minimal` is not equivalent to `is_not_superset`, e.g. purview = (1,2), purviews = [(1,2,3), (1,2), (0)], is_not_superset(purview) returns True, while is_minimal(purview) returns False. In any case I simplified it to:
`def is_not_superset(purview):
      return any((set(purview).issuperset(set(other_purview))) and
                           (set(purview) != set(other_purview)) for other_purview in purviews)`

Reply:
Fixed, except from the logic part because`is_minimal` is not equivalent to `is_not_superset`, e.g. purview = (1,2), purviews = [(1,2,3), (1,2), (0)], is_not_superset(purview) returns True, while is_minimal(purview) returns False.